src/Cli/dotnet/commands/dotnet-workload/list/InstalledWorkloadsCollection.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.NET.Sdk.WorkloadManifestReader;
+
+namespace Microsoft.DotNet.Workloads.Workload.List
+{
+    internal class InstalledWorkloadsCollection : Dictionary<string, string>
```

Comment:
Can we encapsulate instead of inheriting here?  The override to `Add` makes this work differently than I would expect a Dictionary to work.